util/collate/collate.go

```diff
@@ -354,17 +355,17 @@ func CollationToProto(c string) int32 {
 
 // ProtoToCollation converts collation from int32(used by protocol) to string.
 func ProtoToCollation(c int32) string {
-	coll, err := charset.GetCollationByID(int(RestoreCollationIDIfNeeded(c)))
-	if err == nil {
-		return coll.Name
-	}
-	logutil.BgLogger().Warn(
-		"Unable to get collation name from ID, use name of the default collation instead",
-		zap.Int32("id", c),
-		zap.Int("default collation ID", mysql.DefaultCollationID),
-		zap.String("default collation", mysql.DefaultCollationName),
-	)
-	return mysql.DefaultCollationName
+	return CollationID2Name(RestoreCollationIDIfNeeded(c))
+}
+
+// All the names supported collations when new collation is not enabled should be in the following table.
+var supportedCollationWithoutNewCollation = []string{
+	charset.CollationUTF8,
+	charset.CollationUTF8MB4,
+	charset.CollationASCII,
+	charset.CollationLatin1,
+	charset.CollationBin,
+	charset.CollationGBKBin,
```

Comment:
Why not initializing the collations directly instead of names in `init()`?

Reply:
we only need its name here